electron/app/KubeConfigManager.ts

```diff
@@ -8,21 +9,81 @@ import {AlertEnum} from '@shared/models/alert';
 import electronStore from '@shared/utils/electronStore';
 import {getKubeAccess} from '@shared/utils/kubeclient';
 
-import {dispatchToAllWindows} from './ipc/ipcMainRedux';
+import {dispatchToAllWindows, dispatchToFocusedWindow} from './ipc/ipcMainRedux';
+
+const PROXY_PORT_REGEX = /127.0.0.1:[0-9]+/;
 
 let kubeServiceProcess: UtilityProcess | null = null;
+let proxyPort: number | null = null;
+let kubectlProxyProcess: ChildProcessWithoutNullStreams | null = null;
 
-export const startKubeConfigService = () => {
+electronStore.onDidChange('appConfig.kubeConfig', () => {
   if (kubeServiceProcess) {
-    kubeServiceProcess.kill();
+    startKubeConfigService();
+  }
+});
+
+electronStore.onDidChange('appConfig.clusterProxyPort', () => {
```

Comment:
Why are we persisting the clusterProxyPort if the proxy itself will not persist across sessions?